### ISSUE TITLE
MySQL 8.0.28 and TLS versions

### ISSUE THREAD

**Issue by ZeWaren** [mysql-otp/mysql-otp#195]
Hey all,

MySQL 8.0.28 removed the support for TLSv1:
https://dev.mysql.com/doc/refman/8.0/en/encrypted-connection-protocols-ciphers.html#encrypted-connection-protocol-configuration

However, it's still forced as the only version available in the code by default:
https://github.com/mysql-otp/mysql-otp/blob/32327c217f4f785a401e9a3766b22a757f35e4bc/src/mysql_protocol.erl#L432

With the current release, it's mandatory to force the version param to more recent TLS versions:

```
{versions, ['tlsv1.2', 'tlsv1.3']}]}
```
 Otherwise, it fails spectacularly:

```
<0.750.0> exit with reason {failed_to_upgrade_socket,{tls_alert,{protocol_version,"TLS client: In state hello received SERVER ALERT: Fatal - Protocol Version\n"}}} in context child_terminated
2022-03-11 22:50:34.677 [error] <0.754.0> gen_server nexmo_db_poll terminated with reason: {{failed_to_upgrade_socket,{tls_alert,{protocol_version,"TLS client: In state hello received SERVER ALERT: Fatal - Protocol Version\n"}}},{gen_server,call,[test_db,something]}} in gen_server:call/2 line 238
```

I believe this should be addressed or at least be documented.

**Comment by zuiderkwast** [comment]
See #185

**Comment by juhlig** [comment]
Fixed by #185.
